numpy/_core/src/multiarray/number.c

```diff
@@ -328,6 +328,62 @@ array_inplace_matrix_multiply(PyArrayObject *self, PyObject *other)
     return res;
 }
 
+static int
+fast_scalar_power(PyObject *o1, PyObject *o2, int inplace, PyObject **result)
+{
+    if (!PyArray_Check(o1)) {
+        return -1;
+    }
```

Comment:
It seems removing it causes segmentation faults. Perhaps the compiler uses it to optimize?

Reply:
Sorry, i was going to delete it and forgot.  If (and only if) the later exact checks match then the `PyArray_ISOBJECT` can be done safely (although some code paths used to call this with invalid inputs, I don't think they do anymore).

So you can probably remove it, but it would require re-organizing the `PyArray_ISOBJECT` check to the end.
(If it still segfaults, some scalar path calls in here.)

Reply:
Ah, so that it would be memory-safe in case of the nonzero returns! Thanks--this is done.